lib/spack/spack/directives.py

```diff
@@ -424,15 +424,19 @@ def _execute_provides(pkg):
         # to build the ProviderIndex.
         when_spec.name = pkg.name
 
-        for string in specs:
-            for provided_spec in spack.spec.parse(string):
-                if pkg.name == provided_spec.name:
-                    raise CircularReferenceError(
-                        "Package '%s' cannot provide itself.")
+        spec_objs = [spack.spec.Spec(x) for x in specs]
```

Comment:
Honestly I was surprised by seeing this implementation, since neither the fact that multiple specs could go in a single directive nor the fact that multiple specs can be listed in the same string seems to be used in packages. To be clear I couldn't find any directive like:
```python
provides('mpi lapack')
```
or like:
```python
provides('mpi', 'lapack')
```
in any package prior to this PR.